EventEmmiter/eventemmiter.js

```diff
@@ -0,0 +1,68 @@
+class EventEmitter {
+  constructor() {
+    //maintain array with event nae and all listeners which have subscribed to the event
+    this.events = {};
+  }
+
+  subscribe(eventName, callback) {
+    //here our callback is listener
+
+    if (typeof callback !== "function") {
+      throw new Error("Listener must be a function!");
+    }
+
+    //check if event present or not if not
+    //create new event and initialize with empty array
+
+    this.events[eventName] = this.events[eventName] || [];
```

Comment:
you should use object built in method to check if key exists or not instead of this approach.